src/plugin.rs

```diff
@@ -110,30 +136,51 @@ impl MigrationSource<'static> for MigrationList {
 #[command]
 async fn load(
     db_instances: State<'_, DbInstances>,
+    path_mapper: State<'_, fn(String) -> String>,
     migrations: State<'_, Migrations>,
     db: String,
-) -> Result<()> {
-    if !Db::database_exists(&db).await.unwrap_or(false) {
-        Db::create_database(&db).await?;
+) -> Result<String> {
+    #[cfg(feature = "sqlite")]
+    let db = &path_mapper(db);
```

Comment:
This is the same thing i mentioned down below in `initialize`. Going with the `String` directly would remove the `to_string` call at the bottom (which may or may not be a `clone` behind the scenes, idk how the compiler resolves this).